docs/graph-algorithms/breadth-first-search.mdx

```diff
@@ -0,0 +1,341 @@
+---
+title: Breadth First Search
+slug: /graph-algorithms/breadth-first-search
+---
+## Introduction
+
+Breadth-first search (BFS) is an algorithm for traversing or searching tree or graph data structures.It is similar to level-order traversal in trees. It starts at the root node and explores all of the neighbor nodes at the present node before moving on to the nodes at the next depth level.BFS uses queue data structure for traversing the nodes.
+
+
+## Explanation
+
+The BFS traversal of Graphs also traverses the graph in levels. It starts the traversal with a given `source vertex (V)`, visits all of the vertices adjacent to the initially given vertex and `pushes them all to a queue` in order of visiting and `pops an element from the front of the queue`, visits all of its neighbours and pushes the neighbours which are not already visited into the queue and repeats the process until the queue is empty or all of the vertices are visited.
```

Comment:
The theory seems fine but can you change the sentence formation as it sounds a bit vague.